### PR TITLE
Move date.now and console.time audits to table formatter

### PR DESCRIPTION
**Pull request by ebidel**
R: all

This also ads the table heads to the pretty print output. The alignment is not great, but otherwise it's hard to know what the columns mean. Ideally, we'd bring in a CLI table formatter, but ...g3 :(

screen shot 2017-01-26 at 12 51 48 pm

### CODE REVIEW

**Comment by brendankenny** [comment]
>  Ideally, we'd bring in a CLI table formatter, but ...g3

what about something simple like [text-table](https://www.npmjs.com/package/text-table) with no deps and tiny in size.

File: lighthouse-core/formatters/table.js [hunk 1]
```diff
@@ -78,13 +83,16 @@ class Table extends Formatter {
...
     const rows = results.map(result => {
       const cols = headingKeys.map(key => {
+        const val = result[key];
         switch (key) {
           case 'code':
-            return '`' + result[key].trim() + '`';
+            return '`' + val.trim() + '`';
           case 'lineCol':
             return `${result.line}:${result.col}`;
+          case 'isEval':
```
Comment (patrickhulce) [inline lighthouse-core/formatters/table.js]: Not a huge fan of accumulating audit-specific display logic here, do we anticipate `isEval` cropping up in several other places too?

**Comment by patrickhulce** [comment]
thoughts on using `URL.getDisplayName` on those urls if they're not eval? would be nice to standardize there :)

**Comment by paulirish** [comment]
> what about something simple like text-table with no deps and tiny in size.

pretty good. here's their left-right align:

image

**Comment by ebidel** [comment]
I'd like to prettifer the pretty print in another PR, but don't we need a module that's checked in to g3?

**Comment by ebidel** [comment]
There's https://github.com/75lb/table-layout

File: lighthouse-core/formatters/table.js (hunk 1 above)
Comment (ebidel) [inline lighthouse-core/formatters/table.js]: We have a few JS audits that this is relevant for.

**Comment by ebidel** [comment]
@patrickhulce should we `URL.getDisplayName` in the table formatter so everyone gets it and audits don't have to call themselves?

**Comment by brendankenny** [comment]
> I'd like to prettify the pretty print in another PR, but don't we need a module that's checked in to g3?

New modules can be checked in, it's just a big responsibility..."we" have to own compat and maintenance. Since it has no dependencies it makes that a lot easier since it'll only be the addition of a single module.

**Comment by brendankenny** [comment]
> There's https://github.com/75lb/table-layout

yeah, there are a few with a lot more features, but e.g. table-layout brings in seven other modules in addition to itself. I guess it just depends on how much we need fancy table layout.

[cli-table](https://www.npmjs.com/package/cli-table) is another option, and only brings in [colors](https://www.npmjs.com/package/colors), but the compatibility there is harder as it's fixed on 1.03, we already have `~0.6.3` from speedline, and the g3 version is 1.1.2. So someone will have to figure out what changes were made in that range and what incompatibilities exist as a result :)

**Comment by patrickhulce** [comment]
> should we URL.getDisplayName in the table formatter so everyone gets it and audits don't have to call themselves?

I like it in theory, but there are some complexities with inline/previews we have to consider and I was going to suggest accepting a parameter for the base URL so that cross-origin hosts would be shown, since those weren't so obvious for me on a few runs which would complicate the formatter a bit. Either way let's save for another PR?

**Comment by ebidel** [comment]
SG

**Comment by brendankenny** [comment]
table formatter test bugs on travis

**Comment by ebidel** [comment]
fixed tests!

**Pull request merged**